Source/Core/Duality/Launcher/DualityLauncher.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Duality.Backend;
+using Duality.Resources;
+
+namespace Duality.Launcher
+{
+	public class DualityLauncher
+	{
+		public static void Run(string[] args)
+		{
+			System.Threading.Thread.CurrentThread.CurrentCulture = System.Globalization.CultureInfo.InvariantCulture;
+			System.Threading.Thread.CurrentThread.CurrentUICulture = System.Globalization.CultureInfo.InvariantCulture;
+
+			bool isDebugging = System.Diagnostics.Debugger.IsAttached || args.Contains(DualityApp.CmdArgDebug);
+			bool isRunFromEditor = args.Contains(DualityApp.CmdArgEditor);
+			bool isProfiling = args.Contains(DualityApp.CmdArgProfiling);
+
+			// Set up console logging
+			Logs.AddGlobalOutput(new ConsoleLogOutput());
+
+			// Set up file logging
+			StreamWriter logfileWriter = null;
+			TextWriterLogOutput logfileOutput = null;
+			try
+			{
+				logfileWriter = new StreamWriter("logfile.txt");
+				logfileWriter.AutoFlush = true;
+				logfileOutput = new TextWriterLogOutput(logfileWriter);
+				Logs.AddGlobalOutput(logfileOutput);
+			}
+			catch (Exception e)
+			{
+				Logs.Core.WriteWarning("Text Logfile unavailable: {0}", LogFormat.Exception(e));
+			}
+
+			// Set up a global exception handler to log errors
+			AppDomain.CurrentDomain.UnhandledException += CurrentDomain_UnhandledException;
+
+			// Write initial log message before actually booting Duality
+			Logs.Core.Write("Running DualityLauncher with flags: {1}{0}",
+				Environment.NewLine,
+				new[] { isDebugging ? "Debugging" : null, isProfiling ? "Profiling" : null, isRunFromEditor ? "RunFromEditor" : null }.NotNull().ToString(", "));
+
+			// Initialize the Duality core
+			DualityApp.Init(
+				DualityApp.ExecutionEnvironment.Launcher,
+				DualityApp.ExecutionContext.Game,
+				new DefaultAssemblyLoader(),
+				args);
+
+			// Open up a new window
+			WindowOptions options = new WindowOptions
+			{
+				Size = DualityApp.UserData.WindowSize,
+				ScreenMode = isDebugging ? ScreenMode.Window : DualityApp.UserData.WindowMode,
+				RefreshMode = (isDebugging || isProfiling) ? RefreshMode.NoSync : DualityApp.UserData.WindowRefreshMode,
+				Title = DualityApp.AppData.AppName,
+				SystemCursorVisible = isDebugging || DualityApp.UserData.SystemCursorVisible
+			};
+			using (INativeWindow window = DualityApp.OpenWindow(options))
+			{
+				// Load the starting Scene
+				Scene.SwitchTo(DualityApp.AppData.StartScene);
+
+				// Enter the applications update / render loop
+				window.Run();
+
+				// Shut down the Duality core
+				DualityApp.Terminate();
+			}
+
+			// Clean up the log file
+			if (logfileWriter != null)
+			{
+				Logs.RemoveGlobalOutput(logfileOutput);
+				logfileWriter.Flush();
+				logfileWriter.Close();
+				logfileWriter = null;
```

Comment:
Could remove the null assignments, as the program is exiting and they no longer used anyway